docs: add component hub skeleton template + Stack Components section (op-batcher + cannon hubs) (#21865)
ethereum-optimism/optimism: docs/public-docs/op-stack/components/index.mdx

explanation. For release history across all components, see
[Releases](/releases).

Hubs are shipping in reviewed batches. Components without a hub yet are
listed with their release history; their hub pages follow.

## Sequencing

Producing L2 blocks and making their data available.

| Component | What it does | Hub | Releases |
| --- | --- | --- | --- |
| op-batcher | Posts L2 transaction batches to the data availability layer | [Hub](/op-stack/components/op-batcher) | [Releases](/releases/op-batcher) |
| op-conductor | High-availability sequencer coordination service | Planned | [Releases](/releases/op-conductor) |

## Derivation & execution

Deriving the canonical L2 chain from the data availability layer and
executing its blocks.

| Component | What it does | Hub | Releases |
| --- | --- | --- | --- |
| op-node | OP Stack consensus-layer (rollup node) client, in Go | Planned | [Releases](/releases/op-node) |
| kona-node | Rust implementation of the OP Stack rollup node | Planned | [Releases](/releases/kona-node) |
| op-reth | OP Stack execution-layer client built on Reth | Planned | [Releases](/releases/op-reth) |

## Settlement & proofs

> Proposing L2 state to L1 and defending it with fault proofs.

Reviewer: ```suggestion
Proposing L2 state to L1 and proving its validity. 
```
